common/state/state_kv.cpp

```diff
@@ -1274,6 +1753,14 @@ void pdo::state::State_KV::Finalize(ByteArray& outId)
         // output the root id
         outId = rootNode_.GetBlockId();
     }
+    catch (pdo::error::RuntimeError& e)
+    {
+        __on_error__(e.what());
+    }
+    catch (pdo::error::ValueError& e)
+    {
+        __on_error__(e.what());
+    }
     catch(const std::exception& e)
     {
         SAFE_LOG_EXCEPTION("finalize kv");
```

Comment:
why do you need all the distinct catches if you're generating the same error? seems like the catch all would sufficient

Reply:
changed